test-case-reporting/src/test_result_record.ts

```diff
@@ -295,4 +295,35 @@ export class TestResultRecord {
     }));
     /* eslint-enable @typescript-eslint/camelcase */
   }
+
+  /**
+   * Gets a list of the test cases with highest fail/pass/skip/error percentage
+   * @param pageInfo Object with the limit and the offset of the query
+   * @param column Which test status we want to sort by
+   * @param sampleSize The sample size of the computed stats we want
+   */
+  async getTestCasesSortedByStatColumn(
+    {limit, offset}: PageInfo,
+    column: 'passed' | 'failed' | 'skipped' | 'errored',
```

Comment:
Would you prefer I assign `'passed' | 'failed' ...` to a type on test-case-reporting.d.ts?

Reply:
Instead I'd prefer you validate `column` inside the function, throwing a `KeyError` if it's not one of the allowlisted options.

Reply:
Done. There's no KeyError in javascript, so I'm throwing a TypeError, let me know what you think.